src/Disks/IO/ReadBufferFromWebServer.cpp

```diff
@@ -56,15 +56,16 @@ std::unique_ptr<ReadBuffer> ReadBufferFromWebServer::initialize()
     const auto & settings = context->getSettingsRef();
     const auto & server_settings = context->getServerSettings();
 
+    auto connection_timeouts = ConnectionTimeouts::getHTTPTimeouts(settings, server_settings.keep_alive_timeout);
+    connection_timeouts.withConnectionTimeout(std::max<Poco::Timespan>(settings.http_connection_timeout, Poco::Timespan(20, 0)));
+    connection_timeouts.withSslConnectionTimeout(std::max<Poco::Timespan>(settings.http_connection_timeout, Poco::Timespan(20, 0)));
+    connection_timeouts.withReceiveTimeout(std::max<Poco::Timespan>(settings.http_receive_timeout, Poco::Timespan(20, 0)));
```

Comment:
why not chain it?

Reply:
don't know the answer)

That code looks suspicious for me.
We take settings but we do not trust it, we make it sure that it is not too low.
It it ether the settings is wrong and other one has to be used, or its default is wrong.

Reply:
So I don't want to chain them. Just leave it as it -- suspicious.